internal/pkg/cli/svc_deploy.go

```diff
@@ -461,18 +466,32 @@ func (o *deploySvcOpts) deploySvc(addonsURL string) error {
 	return nil
 }
 
-func (o *deploySvcOpts) validateAppVersion(svc *manifest.LoadBalancedWebService) error {
-	var appVersion string
-	var err error
-	if aws.StringValue(svc.Alias) != "" && o.targetApp.RequiresDNSDelegation() {
-		appVersion, err = o.appVersionGetter.Version()
-		if err != nil {
-			return fmt.Errorf("get version for app %s: %w", o.appName, err)
-		}
-		diff := semver.Compare(appVersion, deploy.AliasLeastAppTemplateVersion)
-		if diff < 0 {
-			return fmt.Errorf(`enable "http.alias": the application version should be at least %s`, deploy.AliasLeastAppTemplateVersion)
-		}
+func validateAlias(svc *manifest.LoadBalancedWebService, app *config.Application, envName string, appVersionGetter versionGetter) error {
+	alias := aws.StringValue(svc.Alias)
+	if alias == "" {
+		return nil
+	}
+	if err := validateAppVersion(alias, app, appVersionGetter); err != nil {
+		return err
+	}
+	regEnvHostedZone := regexp.MustCompile(fmt.Sprintf(`^([^\.]+\.)?%s.%s.%s`, envName, app.Name, app.Domain))
```

Comment:
I think we should use [`regexp.Compile`](https://golang.org/pkg/regexp/#Compile) if these are not global variables. Otherwise it could panic.